src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/publishers/TransactionExpiredEventPublisher.kt

```diff
@@ -31,7 +33,28 @@ class TransactionExpiredEventPublisher(
     fun publishExpiryEvents(
         baseTransactions: List<BaseTransaction>,
         batchExecutionTimeWindow: Long
-    ) = publishAllEvents(baseTransactions, TransactionStatusDto.EXPIRED, batchExecutionTimeWindow)
+    ): Mono<Boolean> {
+        // split expired transaction in two lists: one for transactions without requested
+        // authorization and one with requested authorization
+        val splittedTransactions: Map<Boolean, List<BaseTransaction>> =
+            baseTransactions
+                .stream()
+                .collect(Collectors.groupingBy { it is BaseTransactionWithRequestedAuthorization })
+        val baseTransactionsWithRequestedAuthorization = splittedTransactions[true] ?: emptyList()
+        val baseTransactionsActivatedOnly = splittedTransactions[false] ?: emptyList()
```

Comment:
I don't like `splittedTransaction[true]` too much :)
What about:

```suggestion
        val (baseTransactionsWithRequestedAuthorization,  baseTransactionsActivatedOnly) = baseTransactions
                .partition { it is BaseTransactionWithRequestedAuthorization })
```

Using kotlin's [partition](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/partition.html)